sysapps/common/common_api.js

```diff
@@ -80,13 +71,18 @@ var BindingObjectPrototype = function() {
     });
   };
 
-  function addMethodWithPromise(name, Promise) {
+  function addMethodWithPromise(name) {
     Object.defineProperty(this, name, {
       value: function() {
-        var promise_instance = new Promise();
         var args = Array.prototype.slice.call(arguments);
-        this._postMessage(name, args, wrapPromiseAsCallback(promise_instance));
-        return promise_instance;
+        return new Promise(function(resolve, reject) {
```

Comment:
return new Promise((resolve, reject) => error ? reject(error) : resolve(data));

should probably work. If you need to add { } then you need to add 'return'  as well